cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -50,28 +102,409 @@ CDAP supports logging through standard `SLF4J (Simple Logging Facade for Java)
+  retrieve different messages.
+- Log messages of a program can be viewed in the :ref:`CDAP UI <cdap-ui>`.
+
+Program Log File Locations
+--------------------------
+Program logs are stored in locations specified by properties in the
+:ref:`cdap-site.xml <appendix-cdap-default-logging>` file depending on the mode 
+of CDAP (Standalone or Distributed):
+
+- For **Standalone CDAP:** the property ``log.collection.root`` (default
+  ``${local.data.dir}/logs``) is the root location for collecting program logs when in
+  Standalone CDAP.
+
+- For **Distributed CDAP:** the property ``log.base.dir`` (default ``/logs/avro``) is the
+  base directory in HDFS for program logs
+
+.. _configuring-user-application-program-logging:
+
+Configuring Program Logs and Log Levels
+---------------------------------------
+The logging of an application's programs are configured by the ``logback-container.xml``
+file, packaged with the CDAP distribution. This "logback" does log rotation once a day at
+midnight and expires logs older than 14 days. Changes can be made to
+``logback-container.xml``; afterwards, the ``cdap-master`` process will need to be
+restarted for the modified logback file to take effect.
+
+- For **Distributed CDAP:** the ``logback-container.xml`` file is located in ``/etc/cdap/conf``.
+- For **Standalone CDAP:** the file ``logback.xml`` file, located in  ``<cdap-sdk-home>/conf``,
+  configures both "container" and CDAP system service logging.
+  
+You can also use a custom "logback" file with your application, as described in the
+Developers' Manual section :ref:`application-logback`.
+
+Changing Program Log Levels
+---------------------------
+The CDAP :ref:`Logging HTTP RESTful API
+<http-restful-api-logging-changing-program-log-levels>` can be used to set the log levels
+for a particular run of a program. Once changed, they can be reset back to what they
+started with by using the :ref:`reset endpoint <http-restful-api-logging-resetting>`.
+
+To change the log level before an application starts, you can add the logger name as the
+key and log level as the value in the preferences. The logger name should be prefixed with
+``system.log.level``. 
+
+For example, if you want to change the log level of the ``HelloWorld`` class of the
+:ref:`Hello World example <examples-hello-world>` to ``DEBUG``, you would use
+``system.log.level.co.cask.cdap.examples.helloworld.HelloWorld`` as the key and ``DEBUG``
+as the value. This can be applied to any package or classes. If the logger name is
+omitted, it will change the log level of ROOT.
+
+To change the log level of a program dynamically (for a worker, flow, or service which are
+running), see :ref:`Logging HTTP RESTful API
+<http-restful-api-logging-changing-program-log-levels>`. You can change the log level at
+run level of a worker, flow, or service.
+
+**Note:** The log levels can only be changed for programs that are running under
```

Comment:
you mean, before a program starts? Or once it's running? Or both?

I find it a little confusing to speak of "change" in both cases.